x/utils/logging.go

```diff
@@ -17,22 +17,26 @@ func NewLogging() Logging {
 }
 
 // Check logs error -> info, success -> debug
-func (r Logging) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx,
-	next weave.Checker) (weave.CheckResult, error) {
-
+func (r Logging) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx, next weave.Checker) (*weave.CheckResult, error) {
 	start := time.Now()
 	res, err := next.Check(ctx, store, tx)
-	logDuration(ctx, start, res.Log, err, true)
+	var resLog string
+	if err == nil {
```

Comment:
req: this is different behaviour than before as you will miss the res.Log on error results. Please revert

Reply:
I don't agree.

If a handler method (check or deliver) returns an error than the result value is undefined and must not be used. This means that when an error is returned, there is no log message.

@ethanfrey can you confirm the handler return values in case of an error?

Reply:
I get your point but the current [logDuration](https://github.com/iov-one/weave/blob/master/x/utils/logging.go#L50) is very much relying on the ` res.Log`. That is probably a mistake. But still you are changing existing logic. 🦀

Reply:
I have made a change to cleanup `logDuration` and make it clear what is the flow https://github.com/iov-one/weave/pull/531